dev/tools/lib/git.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+/// A wrapper around git process calls that can be mocked for unit testing.
+class Git {
+  Git(this.processManager) : assert(processManager != null);
+
+  final ProcessManager processManager;
+
+  String getOutput(
+    List<String> args,
+    String explanation, {
+    @required String workingDirectory,
+  }) {
```

Comment:
Here and elsewhere, I would separate the cast from the method invocation. maybe a little helper method like:

`String decode(dynamic value) => (value as String).trim();`

but with a better name 😭

Reply:
I can do that. Is your concern line numbers in the stack trace, or is there a higher level code quality concern?

Reply:
Its a bit awkward to read with the cast inline. Another approach would be to use `toString()` instead since that is defined on Object/dynamic

Reply:
implemented `String stdoutToString(dynamic input)`